Core.ApplicationServices/Notification/UserNotificationApplicationService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Core.ApplicationServices.Authorization;
+using Core.DomainModel.Notification;
+using Core.DomainModel.Result;
+using Core.DomainModel.Shared;
+using Core.DomainServices.Authorization;
+using Core.DomainServices.Notifications;
+using System;
+using System.Collections.Generic;
+
+namespace Core.ApplicationServices.Notification
+{
+    public class UserNotificationApplicationService : IUserNotificationApplicationService
+    {
+        private readonly IAuthorizationContext _authorizationContext;
+        private readonly IUserNotificationService _userNotificationService;
+
+        public UserNotificationApplicationService(IUserNotificationService userNotificationService, IAuthorizationContext authorizationContext)
+        {
+            _userNotificationService = userNotificationService;
+            _authorizationContext = authorizationContext;
+        }
+
+        public Result<UserNotification, OperationError> Delete(int id)
+        {
+            var getResult = _userNotificationService.GetUserNotification(id);
+
+            if (!getResult.Ok)
+            {
+                return getResult;
+            }
+
+            var notificationToDelete = getResult.Value;
+
+            if (!_authorizationContext.AllowDelete(notificationToDelete))
```

Comment:
Jeg tror at auth for brugerbeskeder skal begrænses til at tjekke om DU (auth user) er lig med recipient for den besked du henter/sletter